File: 1E1-DM11-Equation-de-Pell-Fermat/calc_sol.py
from math import sqrt

def f(b: int) -> float:
    return sqrt(1 + 5 * b**2)

def test_to(n: int):
    for b in range(1, n+1):
        a = f(b)
        if a.is_integer():
            yield a, b

if __name__ == "__main__":
    for e in test_to(200000000000000):
        print(e)

File: 1E1-DM11-Equation-de-Pell-Fermat/calc_sol.hs
#!/usr/bin/env runhaskell

f b = sqrt (1 + 5 * b^2)

test_to n = [ (a, b) | b <- [1..n]
                     , let a = f b
                     , isNat a
            ]

-- Ignore after
isNat n = n >= 0 && (n - fromIntegral (truncate n)) <= 10^^(-7)

main :: IO()
main = print (map (\(x, y) -> (truncate x, truncate y)) (test_to 200000000000000))
-- to debug change truncate to id